src/Jwt/TokenFactoryInterface.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/*
+ * This file is part of the Mercure Component project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Symfony\Component\Mercure\Jwt;
+
+/**
+ * @author Saif Eddin Gmati <azjezz@protonmail.com>
+ *
+ * @experimental
+ */
+interface TokenFactoryInterface
+{
+    /**
+     * Create a token provider that will  allow publishing
+     * to $publish and subscribing to $subscribe.
+     *
+     * @param string[] $publish
```

Comment:
It may not be obvious what is expected for this argument. Can we add some description? Same for `$subscribe` below?